xbmc/pvr/channels/PVRChannel.cpp

```diff
@@ -184,26 +184,17 @@ void CPVRChannel::ResetEPG()
 
 bool CPVRChannel::UpdateFromClient(const std::shared_ptr<CPVRChannel>& channel)
 {
-  SetClientID(channel->ClientID());
-
   CSingleLock lock(m_critSection);
 
-  if (m_clientChannelNumber != channel->m_clientChannelNumber ||
-      m_strMimeType != channel->MimeType() ||
-      m_iClientEncryptionSystem != channel->EncryptionSystem() ||
-      m_strClientChannelName != channel->ClientChannelName() ||
-      m_bHasArchive != channel->HasArchive())
-  {
-    m_clientChannelNumber = channel->m_clientChannelNumber;
-    m_strMimeType = channel->MimeType();
-    m_iClientEncryptionSystem = channel->EncryptionSystem();
-    m_strClientChannelName = channel->ClientChannelName();
-    m_bHasArchive = channel->HasArchive();
+  SetClientID(channel->ClientID());
+  SetArchive(channel->HasArchive());
 
-    UpdateEncryptionName();
+  m_clientChannelNumber = channel->m_clientChannelNumber;
+  m_strMimeType = channel->MimeType();
+  m_iClientEncryptionSystem = channel->EncryptionSystem();
+  m_strClientChannelName = channel->ClientChannelName();
 
-    m_bChanged = true;
```

Comment:
If the client channel name and number have changed how will this be persisted now?

This will also be true for mime type and encryption system from the client.

Reply:
As far as I can see all these value will never be persisted as they are not even part of the channels table schema.

Reply:
number is stored in channel group member, not channel, group member should be invalidated and persisted properly (I hope).

Reply:
channel name change is handled via SetChannelName() called some lines below, btw.

Reply:
Persisting a group calls IsChanged() on each channel and then persists those values. However if we never set m_bIsChanged that will not happen.

Reply:
As I said, in case of channel name change, m_bChanged will be set for the channel (call to SetChannelName).

For the other values you mentioned, no need to set m_bChanged as those values are never persisted in the db. They are not part of the channels table db schema.

So, what is actually (please proof with a runtime test) not working?

Reply:
Maybe a call to ChannelGroup::Persist() is missing somewhere? But the general logic should be fine.

Reply:
`SetClientChannelNumber()` on `CPVRChannelGroupMember()` is only called when loading from the DB. How and where is client channel number set there?

I don't see anywhere in the code where changed/Needs save is set on it.

Reply:
PersistChannelGroupMembers is only called if the channel has changed.

Reply:
> PersistChannelGroupMembers is only called if the channel has changed.

 where do you see this?

`CPVRChannelGroup::Persist` calls `CPVRDatabase::Persist(CPVRChannelGroup)` calls `CPVRDatabase::PersistGroupMembers` whats wrong with this?

Reply:
So, did a runtime test:

1. In the backend, changed a chanel numer from 18 to 17
2. Started Kodi

=> Kodi detected the change and called SetClientChannelNUmber on the effected groups members

<img width="1028" alt="Screenshot 2021-05-02 at 14 09 46" src="https://user-images.githubusercontent.com/3226626/116812775-c96c8d80-ab50-11eb-88bc-37a1250cbe1a.png">

=> Then, Kodi persisted the changed channel group members

<img width="1330" alt="Screenshot 2021-05-02 at 14 11 01" src="https://user-images.githubusercontent.com/3226626/116812789-d9846d00-ab50-11eb-8c15-ec0061f30c0a.png">

==> seems to work like expected, no?

Reply:
Ok, if you've done the runtime test then clearly I am wrong! Sorry for sending you down rabbit holes for no reason.